examples/event_handling/EventHandling.cpp

```diff
@@ -0,0 +1,338 @@
+                        }
+                        else if constexpr (std::is_same_v<T, sf::Event::MouseButtonPressed>)
+                        {
+                            m_log.emplace_back("Mouse Pressed");
+                        }
+                        else if constexpr (std::is_same_v<T, sf::Event::TouchBegan>)
+                        {
+                            m_log.emplace_back("Touch Began: " + vec2ToString(event.position));
+                        }
+                        else
+                        {
+                            // All unhandled events will end up here
+                            // m_log.emplace_back("Other Event");
+                        }
+                    });
+            }
+            else if (m_handlerType == HandlerType::Forward)
+            {
+                // Forward to other callable
+                // In this case we forward it to our handle member functions
+                // we defined for the concrete event types we want to handle
+                // When choosing this method a default "catch-all" handler must
+                // be available for unhandled events to be forwarded to
+                // If you don't want to provide an empty "catch-all" handler
+                // you will have to make sure (e.g. via if constexpr) that this
+                // lambda doesn't attempt to call a member function that doesn't exist
+                m_window.handleEvents([this](const auto& event) { handle(event); });
+            }
+
+            // Limit the log to 24 entries
+            if (m_log.size() > 24u)
+                m_log.erase(m_log.begin(), m_log.begin() + static_cast<int>(m_log.size() - 24u));
+
+            // Draw the contents of the log to the window
+            m_window.clear();
+
+            for (std::size_t i = 0; i < m_log.size(); ++i)
+            {
+                m_logText.setPosition({50.f, static_cast<float>(i * 20) + 50.f});
+                m_logText.setString(m_log[i]);
+                m_window.draw(m_logText);
+            }
+
+            m_window.draw(m_handlerText);
+            m_window.draw(m_instructions);
+            m_window.display();
+        }
+    }
+
+    // The following handle methods are called by the forwarding event handler implementation
+    // A handle method is defined per event type you want to handle
+    // To handle any other events that are left, the templated handle method will be called
+    // Overload resolution will prefer the handle methods that fit the event type better
+    // before falling back to the templated method
+    void handle(const sf::Event::Closed&)
+    {
+        m_window.close();
+    }
+
+    void handle(const sf::Event::KeyPressed& keyPress)
+    {
+        m_log.emplace_back("Key Pressed: " + sf::Keyboard::getDescription(keyPress.scancode));
+
+        // When the enter key is pressed, switch to the next handler type
+        if (keyPress.code == sf::Keyboard::Key::Enter)
```

Comment:
Personally recommend scancode usage as much as possible, not that it would really matter here

Reply:
Just following the other examples.

Reply:
Maybe the policy should be that we use scancodes if it's layout dependent (e.g. characters), but keycode is fine generic keys like "Return".

Reply:
Sounds reasonable, good thing we don't have a FPS example yet. 😁